src/main/java/com/microsoft/azure/relay/TokenProvider.java

```diff
@@ -50,19 +50,20 @@ public static TokenProvider createSharedAccessSignatureTokenProvider(String keyN
 	 *         generated
 	 */
 	public CompletableFuture<SecurityToken> getTokenAsync(String urlString, Duration validFor) {
-		if (StringUtil.isNullOrEmpty(urlString)) {
-			// TODO: trace
-//            throw RelayEventSource.Log.ArgumentNull(nameof(audience), this);
-		}
-
-		TimeoutHelper.throwIfNegativeArgument(validFor, "validFor");
 		CompletableFuture<SecurityToken> future = new CompletableFuture<SecurityToken>();
-		try {
-			urlString = normalizeAudience(urlString);
-			future = this.onGetTokenAsync(urlString, validFor);
-		}
-		catch (URISyntaxException e) {
+		if (StringUtil.isNullOrEmpty(urlString)) {
+			IllegalArgumentException e = new IllegalArgumentException("Null url provided for security token");
+			RelayLogger.handledExceptionAsWarning(e, this);
```

Comment:
>handledExceptionAsWarning [](start = 15, length = 25)

handledException or throwingException?

Reply:
Should a null token string be a fatal exception?

---
In reply to: [267897192](https://github.com/Azure/azure-relay-java/pull/36#discussion_r267897192) [](ancestors = 267897192)

Reply:
I'm not sure I understand. I think of "handledException" as "someone else threw an exception, I'm catching it, reacting to it, and not re-throwing anything, upstream callers won't see this exception".
I think of "throwingException" as "something went wrong, I'm throwing an exception so upstream callers know something is wrong".

---
In reply to: [267922252](https://github.com/Azure/azure-relay-java/pull/36#discussion_r267922252) [](ancestors = 267922252,267897192)